cranelift/codegen/src/opts.rs

```diff
@@ -149,6 +149,27 @@ impl<'a, 'b, 'c> generated_code::Context for IsleContext<'a, 'b, 'c> {
         self.ctx.func.dfg.value_type(val)
     }
 
+    fn iconst_sextend_etor(&mut self, value: Value) -> Option<(Type, i64)> {
+        let ty = self.value_type(value);
+        if !ty.is_int() {
+            return None;
+        }
+
+        let value = self.ctx.func.dfg.resolve_aliases(value);
+        let ValueDef::Result(inst, 0) = self.ctx.func.dfg.value_def(value) else {
+            return None;
+        };
+        let inst_def = &self.ctx.func.dfg.insts[inst];
+        let InstructionData::UnaryImm {
+            opcode: Opcode::Iconst,
+            imm,
+        } = inst_def
+        else {
+            return None;
+        };
+        Some((ty, self.i64_sextend_imm64(ty, *imm)))
+    }
```

Comment:
I think this might actually be able to get replicated into ISLE with `(extractor iconst_s (iconst ty (i64_sextend_imm64 n)))`, although I forget the precise syntax. Effectively though it should be possible to build up this new extractor based on previous extractors I think?